eZ/Bundle/EzPublishCoreBundle/Templating/Twig/ContextAwareTwigVariablesExtension.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Bundle\EzPublishCoreBundle\Templating\Twig;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use Twig\Extension\AbstractExtension;
+use Twig\Extension\GlobalsInterface;
+
+final class ContextAwareTwigVariablesExtension extends AbstractExtension implements GlobalsInterface
+{
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver
+    ) {
+        $this->configResolver = $configResolver;
+    }
+
+    public function getGlobals(): array
+    {
+        return $this->configResolver->hasParameter('twig_variables')
+            ? $this->configResolver->getParameter('twig_variables')
+            : [];
```

Comment:
What about https://github.com/ezsystems/ezplatform-core/pull/14#discussion_r375439415 ?

Reply:
What's wrong with 3-line ternary? Isn't single `return` cleaner (this is "return this or this" case)?

Reply:
Not sure what's wrong with ternary here? Looks perfectly fine to me 🤔

Reply:
Why not set `[]` as default value in configuration?

Reply:
Done :)